app/scripts/lib/errors.js

```diff
@@ -132,13 +132,16 @@ define([
         return this.toError('SERVICE_UNAVAILABLE');
       }
 
-      var errObj = xhr.responseJSON;
+      var serverError = xhr.responseJSON;
 
-      if (! errObj) {
+      if (! serverError) {
         return this.toError('UNEXPECTED_ERROR');
       }
 
-      return this.toError(errObj.errno);
+      var normalizedError = this.toError(serverError.errno);
+      normalizedError.code = serverError.code;
```

Comment:
Can you add a comment above this line about what the code represents and why the code is transferred across?